upath/tests/test_pydantic.py

```diff
@@ -0,0 +1,78 @@
+import json
+from os.path import abspath
+
+import pydantic
+import pydantic_core
+import pytest
+
+from upath import UPath
+
+
+@pytest.mark.parametrize("source", ["json", "python"])
+def test_validate_from_str(source):
+    input = f"file://{abspath('/my/path')}"
+
+    ta = pydantic.TypeAdapter(UPath)
+    if source == "json":
+        output = ta.validate_json(json.dumps(input))
+    else:  # source == "python"
+        output = ta.validate_python(input)
+
+    assert isinstance(output, UPath)
+    assert str(output) == input
```

Comment:
Keep input the same as before, and wrap both sides on the assert in os.path.abspath. That represents the intend of the test better.